### ISSUE TITLE
XSetInputFocus called before window is viewable.

### ISSUE THREAD

**Issue by mpatraw** [SFML/SFML#991]
I'm running
Linux zork 4.2.3-1-ARCH #1 SMP PREEMPT Sat Oct 3 18:52:50 CEST 2015 x86_64 GNU/Linux
compiling HEAD against this code

```cpp
#include <SFML/Window.hpp>

constexpr int width = 640;
constexpr int height = 480;
constexpr const char *title = "test";

int main(int argc, char *argv[])
{
	auto style = sf::Style::Titlebar | sf::Style::Close;
	sf::Window window(sf::VideoMode(width, height), title, style);

	while (window.isOpen()) {
		sf::Event event;
		while (window.pollEvent(event)) {
			if (event.type == sf::Event::Closed) {
				window.close();
			}
		}
	}

	return 0;
}
```

and I get the output

```
X Error of failed request:  BadMatch (invalid parameter attributes)
  Major opcode of failed request:  42 (X_SetInputFocus)
  Serial number of failed request:  77
  Current serial number in output stream:  86
```

My graphics card is an Intel Atom Z36xxx/Z37xxx.

**Comment by eXpl0it3r** [comment]
Please refer to the (Contribution Guidelines](http://www.sfml-dev.org/contribute.php) in the future and post to the forum first instead. We try to keep the issue tracker for confirmed bugs and features only.

What's your window manager?

**Comment by mpatraw** [comment]
Thanks. My window manager is dwm-6.0.

**Comment by kirbyUK** [comment]
Confirming this issue with the same code posted above, and on dwm-6.0 and Intel graphics of some description.

**Comment by treere** [comment]
problem also in dwm-6.1 . SFML 2.2 works

**Comment by Palesand** [comment]
Same issue under dwm, smells like this similiar issue in sdl with bad assumptions about windowmanager behaviour: https://bugzilla.libsdl.org/show_bug.cgi?id=2997

**Comment by hdclark** [comment]
I can confirm this bug is present in the latest release on Arch Linux with dwm. Observed with all combinations of the following relevant installations: official Arch sfml package (2.3.2-3), AUR 'sfml-git' package (2.3.2.r38.g1217699-1), dwm 6.0-1, dwm (6.1-2), and xorg-server (1.18.0-3).

The problem can be observed with the SFML examples provided in /usr/share/SFML/examples/ . The X11 example runs fine, but the OpenGL example dies immediately after the window is created:

    [hal@machine examples]$ ./X11/X11 
    [hal@machine examples]$ ./opengl/opengl 
    X Error of failed request:  BadMatch (invalid parameter attributes)
      Major opcode of failed request:  42 (X_SetInputFocus)
      Serial number of failed request:  80
      Current serial number in output stream:  89

Reverting to SFML 2.2 at this point would be especially painful owing to the recent GCC C++ ABI change.

**Comment by hdclark** [comment]
By the way, a workaround is to recompile SFML 2.2 using the Arch ABS. Simply modify the latest sfml package's PKGBUILD pkgver variable from 2.3 to 2.2. The bug is absent in 2.2.

**Comment by OvaUbe** [comment]
Same on Arch Linux with xmonad-0.12.

**Comment by binary1248** [comment]
Try out the [feature/grab_mouse branch](https://github.com/SFML/SFML/tree/feature/grab_mouse) commit 7c838fa17a376ae91b8b295f3d94f614848c1548. It should have fixed this problem.

**Comment by x421** [comment]
The same for ratpoison wm :cry:

**Comment by binary1248** [comment]
I think by now it is kind of obvious that many people are able to reproduce the issue. At this stage, it would help more to test the proposed fix.

**Comment by x421** [comment]
The fix works. :+1:

**Comment by eXpl0it3r** [comment]
Fixed in 6f3273b7a6d529fd7902a1f6e2de9d0398f2cde1
